File: Day6/day6.py
'''
    PART 1
'''
# Simple function to check if all characters are unqiue
def check(str: str) -> bool:
    return len(set(list(str))) == len(str)

def __main__():

    part2 = True

    # read in data from input file
    data = ''
    with open('./day6.in', 'r') as file:
        data = file.readlines()[0]

    # iterate through characters in data stream
    i = 0
    for _ in data:
        # get next four characters
        read_data = data[i:4+i if not part2 else 14+i]

        i+=1
        # check if all are unique
        if check(read_data):
            # if so, print the value for the 
            # end of the flag and return
            print(i+3 if not part2 else 13+i)
            return

if __name__ == "__main__": __main__()

File: Day6/day6.js
const fs = require('fs')

const check = (str) => {
    return new Set(str).size === str.length
}

fs.readFile('./day6.in', 'utf-8', (err, data) => {
    let stream = ''
    for(let i = 0; i < data.length; i++) {
        stream = data.slice(i, i+4)
        if(check(stream)) {
            console.log(i+4)
            break
        }
    }
})